classes/local/util/manager.php

```diff
@@ -98,6 +98,45 @@ public function make() {
     }
 
     /**
+     * Create the plugin files at $targetdir.
+     *
+     * @param string $targetdir The target directory for the files.
+     */
+    public function write_files($targetdir) {
+
+        if (empty($this->files)) {
+            throw new exception('There are no files to write');
+        }
+
+        $result = mkdir($targetdir, 0755, true);
+        if ($result === false) {
+            throw new exception('Error creating target directory: '.$targetdir);
+        }
+
+        foreach ($this->files as $filename => $file) {
+
+            $filepath = $targetdir.'/'.$filename;
+            $dirpath = dirname($filepath);
+
+            if (!file_exists($dirpath)) {
+                $result = mkdir($dirpath, 0755, true);
+                if ($result === false) {
+                    throw new exception('Error creating directory: '.$dirpath);
+                }
+            }
+
+            $filehandle = fopen($filepath, 'w');
```

Comment:
Just wondering, is there a reason why not to use simple `file_put_contents()` ? http://php.net/manual/en/function.file-put-contents.php